pytorch_lightning/utilities/model_summary.py

```diff
@@ -261,16 +261,10 @@ def _forward_example_input(self) -> None:
         input_ = model.example_input_array
         input_ = model._apply_batch_transfer_handler(input_)
 
-        if (
-            trainer is not None
-            and trainer.amp_backend == AMPType.NATIVE
-            and trainer._device_type != _AcceleratorType.TPU
-        ):
-            model.forward = torch.cuda.amp.autocast()(model.forward)
```

Comment:
is this a bug in the current code that we patch the `forward` without ever resetting it?

Reply:
maybe not since eventually all the steps will be happening on precision=16